### ISSUE TITLE
Recreate configuration after cluster has been recreated

### ISSUE THREAD

**Issue by bo0ts** [rabbitmq/messaging-topology-operator#624]
Currently the operator does not notice when a cluster referenced by topology objects is deleted and recreated. This leads to a situation where topology objects show up as "Ready" while they are not. It would be nice if the operator could either 

* periodically (or on rabbitmqcluster events) check the state of the object and act accordingly
* watch create/delete events on a referenced rabbitmqcluster and recreate the objects

**Comment by bo0ts** [comment]
I'm aware of that feature, but that seems like a poor workaround for just having the topology operator getting notified on rabbitmq cluster changes. Why not just add a `Watches(...)` statement for the `TopologyReconciler`?

**Comment by Zerpet** [comment]
I understand that the preferred outcome is to add a `Watch()` in the topology operator to react to update/delete events on `RabbitmqCluster` objects. What should happen when a `RabbitmqCluster` object is deleted?

I'm inclined to delete cascade topology objects, because the operator doesn't know if the rabbitmq cluster is coming back. For updates, we can run the current reconcile logic.

What do you think @VannTen ?

**Comment by bo0ts** [comment]
@Zerpet I think you don't interpret ownership in the usual way of kubernetes objects. When I create a `vhosts.rabbitmq.com` that object is controlled by the topology operator and I don't know of any case where a controller actually deletes the resource it is controlling.

**Comment by VannTen** [comment]
I agree with @bo0ts , the operator should not delete the objects , just reconcile them again.
It does not own the object in the ownerRefs sense of kubernetes, it merely implements their semantics.
